### PR TITLE
Fix issue 3424 - Fix indent in XLIFF output

### PR DESCRIPTION
**Pull request by mastizada**

### CODE REVIEW

**Comment by mastizada** [comment]
Tested in pontoon project, in {file} item `<trans-unit id="Add to Bookmarks">`: `note` element was inlined with `target` element, after adding these changes indent solved.
`file`: https://raw.githubusercontent.com/mastizada/firefoxios-l10n/master/az/firefox-ios.xliff

File: translate/storage/lisa.py [hunk 1]
```diff
@@ -302,8 +302,24 @@ def addunit(self, unit, new=True):
         if new:
             self.body.append(unit.xmlelement)
 
+    def cleanindent(self, elem, level=0):
+        i = "\n" + level*"  "
+        if len(elem):
+            if not elem.text or not elem.text.strip():
+                elem.text = i + "  "
```
Comment (unho) [inline translate/storage/lisa.py]: Altering the text in the elements might have unintended consequences.

**Comment by unho** [comment]
@mastizada I have several concerns about this PR:

- Lack of tests
- It changes not only XLIFF but other XML formats
- It changes the indent for all the tags while it looks like the idea is to just fix indent for `note` tag
- The indent amount is hardcoded

Can you please address these?

**Comment by BhaaLseN** [comment]
Thats basically what I was trying to achieve with #3611, but it still fails in some cases (at least I randomly get misaligned text in our VCS after Weblate touches the files).

However, I _do_ want to point out that this could be useful for all XML formats (not just XLIFF). ResX might benefit from this as well (especially if we can get rid of the terrible inline `tail` tracking/modification), plus a generic flat XML format I'm currently working on (where I'm basically stuffing a method similar to this one in before running `serialize`, albeit mine is called `reindent` instead of `cleanindent`)

A good place for such a method could be `translate.misc.xml_helpers` to decouple it from the actual XML formats.

**Comment by nijel** [comment]
I've tried to write something more generic based on this in https://github.com/translate/translate/pull/3871

**Comment by unho** [comment]
Closing this because issue https://github.com/translate/translate/issues/3424 was fixed by PR https://github.com/translate/translate/pull/3871

@mastizada Thanks for your effort anyway!

**Comment by mastizada** [comment]
Thanks @unho , any ideas when it will be published? (current version is 2.3.1)

**Comment by unho** [comment]
@mastizada I would like some other PRs to be merged first. After that we might do a release. I would personally like a release to be done in like a month, but no promises.

**Pull request closed without merging**